### ISSUE TITLE
Aria Roles Missing Proper Element Support Structures

### ISSUE THREAD

**Issue by psenger** [moroshko/react-autosuggest#271]
Hello Misha, 

Firstly, thank you for producing this product, it is very useful. I would like to raise this issue, talk to you about it, fix it, and then create a pull-request.
### Background

Having recently been sued for a lack of accessibility, my client decided to re-write their flag-ship product in ReactJS with this component. Assigned the task of making our site compliant and the liaison with the accessibility tester, I was left surprised by a report about this component. 

To my ignorance, I must confess, I do not need the use of OSX Voice Over, Jaws, nor any other visually impaired software. However, I use it for testing and I'm still learning how to use it. In addition, I use a basic auditing tools in Chrome called "Accessibility Developer Tools" located here [https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb). This reporting tool is limited by its inability to properly evaluate the context of the information and interact with a dynamic website. However, the following error created by the "Accessibility Developer Tools" tool led me into investigating the error further.
### Error

_Elements with ARIA roles must ensure required owned elements are present_. See:
[ax_aria_08](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_08)
### Replication

This error can easily be replicated on the [demo site](http://react-autosuggest.js.org/), by installing the Chrome Extension [Accessilibity Developer Tools](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb), opening the chrome developer tools, clicking on Audits, selecting Accessibility, and running the report.
### Meaning of the error

Referring to the W3 [Roles their Relationship and requirements](https://www.w3.org/TR/wai-aria/roles#role_definitions)

I believe the role "combobox" acts as composite user interfaces and requires the presence of other roles in the DOM, even if they do not contain data. In the following example we see a combo without data. I think the error occurs specifically when the combo box is empty. Because react-autosuggest does not put an empty element in the DOM.

EG:

``` html
<input type="text" aria-label="Tag" role="combobox" aria-expanded="true"
  aria-autocomplete="list" aria-owns="owned_listbox" aria-activedescendant="selected_option">
<ul role="listbox" id="owned_listbox">
<!-- insert li element of selectable items -->
</ul>
```
### What I would like to do

If my findings are wrong, please tell me so. If Im right, I would like to alter the code to add a container to the DOM and inject or delete the dynamic content into the container.

Sincerely,
[Philip A Senger](https://github.com/psenger)

**Comment by moroshko** [comment]
@psenger Thanks a lot for all this information!

At the moment, the following HTML is generated:
```html
<div class="react-autosuggest__container">
  <input role="combobox">
  <div class="react-autosuggest__suggestions-container">
    <ul role="listbox">
      <li role="option">
    </ul>
  </div>
</div>
```

and `<ul>` element is there only when suggestions are presented.

Quick test shows that, if we move `role="listbox"` from the `<ul>` to its parent `<div>`, the error disappears.

But, I'm not convinced that it's a good idea to do that, because then [the following accessibility requirement](https://www.w3.org/TR/wai-aria/roles#listbox) breaks:
> List boxes contain children whose role is option.

I also quickly checked what Google, Bing, and Facebook autocompletes do, and they all have the same error:
> Elements with ARIA roles must ensure required owned elements are present

At this point, I'm not sure what should we do. Maybe some accessibility experts can recommend something?

**Comment by psenger** [comment]
I was told we will have access to an accessibility expert shortly. I will ask them specifically about this and respond to this thread with their recommendations.

**Comment by moroshko** [comment]
@psenger Did you get any recommendations?

**Comment by moroshko** [comment]
Closing for now. Please free to reopen if you have any recommendation with reputable pointers.
